### ISSUE TITLE
Following docs for Dynamic supervisor does not work

### ISSUE THREAD

**Issue by dan-cooke** [elixir-lang/elixir-lang.github.com#1770]
In particular the issue seems to be that using a name in `DynamicSupervisor.start_child` does not work. I did find this odd as previously we had always passed the PId of the running process.

[Anyway heres the section that is not working for me
](https://hexdocs.pm/elixir/dynamic-supervisor.html#the-bucket-supervisor)
```.exs
 def init(:ok) do
    children = [
      {KV.Registry, name: KV.Registry},
      {DynamicSupervisor, name: KV.BucketSupervisor, strategy: :one_for_one}
    ]

    Supervisor.init(children, strategy: :one_for_one)
  end

```
Running

```.exs

{:ok, bucket} = DynamicSupervisor.start_child(KV.BucketSupervisor, KV.Bucket)

```

Results in 
```
 DynamicSupervisor.start_child(KV.BucketSupervisor, KV.Bucket)
** (exit) exited in: GenServer.call(KV.BucketSupervisor, {:start_child, {{KV.Bucket, :start_link, [[]]}, :permanent, 5000, :worker, [KV.Bucket]}}, :infinity)
    ** (EXIT) no process: the process is not alive or there's no process currently associated with the given name, possibly because its application isn't started
    (elixir 1.18.0-dev) lib/gen_server.ex:1121: GenServer.call/3
    iex:1: (file)

```

## Repo
Here is a [reproduction repo](https://github.com/dan-cooke/elixir-issue)

Its potentially an error on my part, but Im 99% sure I have followed the docs exactly

I've cross-referenced with[ the example repo ](https://github.com/josevalim/kv_umbrella) (last updarted 7 years ago) so I would imagine there has been a slight change in the elixir API since then?

I have noticed a few other inconsitencies while going through the Mix OTP docs. 

For example, the return of `start_link` is constantly referenced as a tuple of `{:ok, pid}` but it seems that it is now just pid? 

## Versions
Erlang/OTP 27 [erts-15.1] [source] [64-bit] [smp:32:32] [ds:32:32:10] [async-threads:1] [jit:ns]

Elixir 1.18.0-dev (9d5af3f) (compiled with Erlang/OTP 27)

Arch Linux

**Comment by dan-cooke** [comment]
Okay this has magically just started working. The only thing that I can think of is that I installed erlang via `pacman -S erlang` so I could debug the applications tree

It could have been some issue with that I imagine

**Comment by sabiwara** [comment]
Thank you @dan-cooke for providing a sample repo.
I was also unable to reproduce the issue even when using 1.18.0-dev:

<img width="643" alt="Screenshot 2024-10-05 at 19 49 34" src="https://github.com/user-attachments/assets/7f32030f-eb44-41b3-ab62-69f85041325b">

> For example, the return of start_link is constantly referenced as a tuple of {:ok, pid} but it seems that it is now just pid?

I will try to go through the guide again, any specific part where you noticed such an issue?

**Comment by dan-cooke** [comment]
@sabiwara thanks looking into this, but again I cannot seem to reproduce the earlier issue I was having either. 

I am 100% certain that when I ran 
```.ex
GenServer.call(registry, {:lookup, "shopping"})
```

that it returned just a PID, and not the tuple with `{:ok, PID}` 

I am so certain, because I found that very strange that my Elixir was behaving differently than the documentation. 

I honestly have no idea why it has now changed, but the only thing that I can think of is something to do wtih not having the "right" version of erlang installed? 

Just one of those weird mysteries, thanks anyway!
